File: apps/mobile/lib/domain/entities/role.dart
class RoleEntity {
  final String id;
  final String name;
  final String description;

  const RoleEntity({
    required this.id,
    required this.name,
    required this.description,
  });

  @override
  String toString() {
    return 'RoleEntity(id: $id, name: $name, description: $description)';
  }
}

File: services/api/src/domain/models/user/Role.ts
import mongoose, { Schema, Document, Types } from 'mongoose';

export interface IRole extends Document {
  _id: Types.ObjectId;
  name: 'SuperAdmin' | 'SchoolAdmin' | 'Teacher' | 'Student' | 'CanteenStaff' | 'Nurse' | 'Security';
  description?: string;
  permissions?: string[];
}


const RoleSchema = new Schema<IRole>(
  {
    _id: { type: Schema.Types.ObjectId, auto: true },
    name: {
      type: String,
      enum: [
        "SuperAdmin",
        "SchoolAdmin",
        "Teacher",
        "Student",
        "CanteenStaff",
        "Nurse",
        "Security",
      ],
      required: true,
      unique: true,
      index: true, //thêm index
    },
    description: { type: String },
    permissions: [{ type: String }],
  },
  { timestamps: true }
);

export default mongoose.model<IRole>("Role", RoleSchema);

